src/utils/packageUtils.ts

```diff
@@ -516,3 +516,28 @@ export function copyDescriptorProperties(
 export function replaceIfEmpty<T>(value: T, replacement: T): T {
   return !isEmpty(value) ? value : replacement;
 }
+
+/**
+ * Brand new SFDX projects contain a force-app directory tree contiaining empty folders
+ * and a few .eslint.json files. We still want to consider such a directory tree
+ * as 'empty' for the sake of operations like downloading package version metadata.
+ *
+ * @param directory The absolute path to a directory
+ * @returns true if the directory contains nothing except empty directories or
+ * directories containing only an .eslint.json file.
+ */
+export function isPackageDirectoryEffectivelyEmpty(directory: string): boolean {
+  if (!fs.lstatSync(directory).isDirectory()) {
+    return false;
+  }
+  const entries = fs.readdirSync(directory, { withFileTypes: true });
+  let isEffectivelyEmpty = true;
+  entries.forEach((entry) => {
+    if (entry.isDirectory()) {
+      isEffectivelyEmpty = isPackageDirectoryEffectivelyEmpty(join(directory, entry.name));
+    } else if (entry.name !== '.eslint.json') {
+      isEffectivelyEmpty = false;
+    }
+  });
+  return isEffectivelyEmpty;
```

Comment:
nit: using `entries.every()` would short circuit faster with a bit less code. Lines 534-542 could be:
```typescript
return entries.every((entry) =>
  entry.isDirectory() ? isPackageDirectoryEffectivelyEmpty(join(directory, entry.name)) : entry.name === '.eslint.json')
);
```